kale/embed/factorization.py

```diff
@@ -267,80 +269,123 @@ def inverse_transform(self, x):
 
 class MIDA(BaseEstimator, TransformerMixin):
     """Maximum independence domain adaptation
+
     Args:
         n_components (int): Number of components to keep.
-        kernel (str): "linear", "rbf", or "poly". Kernel to use for MIDA. Defaults to "linear".
+        kernel (str): Kernel to use for MIDA. Defaults to "linear".
+        gamma (float): Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other kernels. If ``gamma`` is
+            ``None``, then it is set to ``1/n_features``., default=None.
+        degree (float): Degree for poly kernels. Ignored by other kernels., default=3
+        coef0 (float): Independent term in poly and sigmoid kernels.Ignored by other kernels. Default=1
         mu (float): Hyperparameter of the l2 penalty. Defaults to 1.0.
         eta (float): Hyperparameter of the label dependence. Defaults to 1.0.
-        augmentation (bool): Whether using covariates as augment features. Defaults to False.
+        augmentation (bool): Whether using domain/group labels as augment features. Defaults to False.
+        fit_label (bool): Whether to fit the label for semi-supervised MIDA. Defaults to False.
         kernel_params (dict or None): Parameters for the kernel. Defaults to None.
 
     References:
         [1] Yan, K., Kou, L. and Zhang, D., 2018. Learning domain-invariant subspace using domain features and
             independence maximization. IEEE transactions on cybernetics, 48(1), pp.288-299.
     """
 
+    _parameter_constraints: dict = {
+        "n_components": [
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "kernel": [
+            StrOptions({"linear", "poly", "rbf", "sigmoid", "cosine", "precomputed"}),
+            callable,
+        ],
+        "gamma": [
+            Interval(Real, 0, None, closed="left"),
+            None,
+        ],
+        "degree": [Interval(Real, 0, None, closed="left")],
+        "coef0": [Interval(Real, None, None, closed="neither")],
+        "kernel_params": [dict, None],
+        "mu": [Real],
```

Comment:
constraint both `mu` and `eta` to be strictly positive like in the original paper (Yan et al., 2017).

<img width="819" alt="image" src="https://github.com/user-attachments/assets/e7502c4e-7d0b-4142-8384-c531678c400e">

`eta=gamma` in this case
<img width="779" alt="image" src="https://github.com/user-attachments/assets/11daa768-558b-4d10-a27a-4af10feaf170">